internal/lifecycle/lifecycle.go

```diff
@@ -40,9 +41,21 @@ func (l Lifecycle) Execute(commandsDir string) error {
 	return nil
 }
 
-// NewLifecycle returns a lifecycle built with reasonable defaults based on whether the sample is java-based
+// NewLifecycle tries to parse the different options provided for build and deploy command configuration. If none of
+// those options are set up, it falls back to reasonable defaults based on whether the sample is java-based
 // (has a pom.xml) that doesn't have a Dockerfile or isn't.
 func NewLifecycle(sampleDir, serviceName, gcrURL string) Lifecycle {
+	readmePath := fmt.Sprintf("%s/README.md", sampleDir)
+
+	if _, err := os.Stat(readmePath); err == nil {
+		lifecycle, err := parseREADME(readmePath, serviceName, gcrURL)
+		if err == nil {
+			log.Println("Using build and deploy commands found in README")
+			return lifecycle
+		}
+		fmt.Printf("Failed parsing README: %v\n", err)
```

Comment:
I would think it would be preferable to have a fatal error here if the README exists but is invalid, rather than continuing to the backup methods.

Reply:
Resolved. Now we're only continuing to backup methods when either (a) the README.md file doesn't exist in the root directory or (b) it's obvious that the user hasn't tried to set the README up to work with our program (there are no tagged code blocks). This is per @kurtisvg suggestion in chat.